ballerina-shell/modules/shell-core/src/main/java/io/ballerina/shell/parser/ParserConstants.java

```diff
@@ -42,5 +45,16 @@ public static boolean isFunctionNameRestricted(String functionName) {
         return RESTRICTED_FUNCTION_NAMES.contains(functionName) || functionName.startsWith(WRAPPER_PREFIX);
     }
 
+    /**
+     * Check if the given source includes restricted keywords
+     * @param source
+     * @return {@code Optional<String>} if the source include restricted keywords. {@code Optional<Empty>} otherwise
+     */
+    public static Optional<String> getMatchedRestrictedSource(String source) {
+        return RESTRICTED_SOURCE_NAMES.stream()
+                .filter(source::contains)
+                .findFirst();
```

Comment:
```suggestion
        return RESTRICTED_SOURCE_NAMES.contains(source);
```

Reply:
Hi @LakshanWeerasinghe,

Thanks for the review. This was my first approach as well, returning a boolean value. However, if we were to return a boolean value we wouldn't be able to provide a detailed error message explaining why the input is rejected, as in  [this code block](https://github.com/ballerina-platform/ballerina-lang/pull/43058/files#diff-c7cfdebbd1741dc7606c90eeeb71bf67db2686e5243609cce468d04fd0c3dc9fR145-R146). What do you think?

See below for more reference.

> Found restricted keyword '_man()_' in the declaration.

![bal-shell-stackoverflow-main-fix](https://github.com/ballerina-platform/ballerina-lang/assets/39789194/5a3471b8-2da4-4e15-80b0-0f2420985f5e)